### PR TITLE
Add linux nodeSelector to all Helm jobs

### PR DESCRIPTION
**Pull request by brandond**
For:
* rancher/rke2#1683

### CODE REVIEW

File: pkg/helm/controller.go [hunk 1]
```diff
@@ -279,15 +284,19 @@ func job(chart *helmv1.HelmChart) (*batch.Job, *core.ConfigMap, *core.ConfigMap)
 		})
 	}
 
+	job.Spec.Template.Spec.NodeSelector = make(map[string]string)
+	job.Spec.Template.Spec.NodeSelector[core.LabelOSStable] = "linux"
```
Comment (briandowns) [inline pkg/helm/controller.go:288]: Does the k8s package have a const for this value?

Comment (brandond) [reply pkg/helm/controller.go:288]: No, just lots of string literals like this one.

Comment (briandowns) [reply pkg/helm/controller.go:288]: Darn, was hoping for something like they have for namespaces. Oh well. Thanks!

**Review by briandowns — APPROVED** [review]

**Review by dweomer — APPROVED** [review]
LGTM (Looks Gross To Me)

**Pull request merged**